flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/join/JoinUtil.scala

```diff
@@ -199,31 +157,38 @@ object JoinUtil {
     val generator = new CodeGenerator(
       config,
       nullCheck,
-      returnType)
+      leftType,
+      Some(rightType))
+
+    val conversion = generator.generateConverterResultExpression(
+      returnType.physicalTypeInfo,
+      returnType.physicalType.getFieldNames)
 
     // if other condition is null, the filterfunc always return true
     val body = if (otherCondition == null) {
```

Comment:
@fhueske @hongyuhong - i think it is Lietral(True)...at least when i was working with inner queries i was always testing for an always true condition